internal/internal_task_handlers.go

```diff
@@ -664,12 +665,12 @@ func (wth *workflowTaskHandlerImpl) getOrCreateWorkflowContext(
 	historyIterator HistoryIterator,
 ) (workflowContext *workflowExecutionContextImpl, err error) {
 	metricsScope := wth.metricsScope.GetTaggedScope(tagWorkflowType, task.WorkflowType.GetName())
-	defer func() {
+	defer func(scope tally.Scope) {
```

Comment:
probably fine but: since we _intentionally_ do not want to use the (possibly mutated) outer-closure's `metricsScope`, personally I'd vote to shadow the name here to make it totally inaccessible.

Reply:
addressed